src/engine/sparqlExpressions/PrefilterExpressionIndex.cpp

```diff
@@ -489,6 +500,80 @@ ValueId PrefilterExpression::getValueIdFromIdOrLocalVocabEntry(
                     referenceValue);
 }
 
+// SECTION PREFIX-REGEX
+//______________________________________________________________________________
+std::unique_ptr<PrefilterExpression> PrefixRegexExpression::logicalComplement()
+    const {
+  return make<PrefixRegexExpression>(prefix_, !isNegated_);
+};
+
+//______________________________________________________________________________
+bool PrefixRegexExpression::operator==(const PrefilterExpression& other) const {
+  const auto* otherPrefixRegex =
+      dynamic_cast<const PrefixRegexExpression*>(&other);
+  if (!otherPrefixRegex) {
+    return false;
+  }
+  return isNegated_ == otherPrefixRegex->isNegated_ &&
+         prefix_ == otherPrefixRegex->prefix_;
+};
+
+//______________________________________________________________________________
+std::unique_ptr<PrefilterExpression> PrefixRegexExpression::clone() const {
+  return make<PrefixRegexExpression>(*this);
+};
+
+//______________________________________________________________________________
+std::string PrefixRegexExpression::asString(
+    [[maybe_unused]] size_t depth) const {
+  return absl::StrCat(
+      "Prefilter PrefixRegexExpression with prefix ", prefix_,
+      ".\nExpression is negated: ", isNegated_ ? "true.\n" : "false.\n");
+};
+
+//______________________________________________________________________________
+BlockMetadataRanges PrefixRegexExpression::evaluateImpl(
+    const Vocab& vocab, const ValueIdSubrange& idRange,
+    BlockMetadataSpan blockRange) const {
+  static_assert(Datatype::LocalVocabIndex > Datatype::VocabIndex);
+  static_assert(Vocab::PrefixRanges::Ranges{}.size() == 1);
+  using LVE = LocalVocabEntry;
+  LocalVocab localVocab{};
+  std::string_view prefixSv(&prefix_.front(), prefix_.size());
+  const auto& prefixRanges = vocab.prefixRanges(prefixSv);
```

Comment:
Does this work?  I think currently `vocab.prefixRanges` requires the value to start with a `"` etc.

Reply:
(and I am not sure in which cases this is present and in which cases not).